### PR TITLE
Tests that all flags at addr are shown

### PR DESCRIPTION
**Pull request by kazarmy**
Tests for radare/radare2#7465.

### CODE REVIEW

**Comment by kazarmy** [comment]
Shouldn't the `section_end.` flags be at the address of the section's last byte instead of last byte address + 1?

**Comment by radare** [comment]
example?

**Comment by radare** [comment]
end is at begin+size

**Comment by kazarmy** [comment]
Example is at https://github.com/kazarmy/radare2-regressions/blob/ff643f084fe563a50cf7b57848b10ea3ba625227/t/cmd_pd#L649. The reason there are 2 section-related flags is because for the test binary used, the section flags are:
```
[0x0040149e]> f~section
0x00401000 4096 section..text
0x00402000 0 section_end..text
0x00402000 4096 section..rdata
0x00403000 0 section_end..rdata
... [snipped since not relevant]
```

**Comment by kazarmy** [comment]
> end is at begin+size

Ok.

**Comment by kazarmy** [comment]
> Why remove that test

It's ... the test is back since it provides the motivation for the PR in the first place.

File: t/cmd_pd [hunk 1]
```diff
@@ -707,6 +711,23 @@ EXPECT='            0x140001004      488d05f54f01.  lea rax, str._tANSI_r_n    ;
...
+NAME='multiple flags at addr'
+FILE=../bins/pe/testapp-msvc64.exe
+ARGS=
+CMDS='e asm.filter=false
+e asm.cmtright=true
+pd 1 @ 0x140001004
+e asm.cmtright=false
+pd 1 @ 0x140001004
+'
+EXPECT='            0x140001004      488d05f54f01.  lea rax, 0x140016000       ; str._tANSI_r_n ; section..data ; ".ANSI.."
```
Comment (radare) [inline t/cmd_pd]: this looks like trash. dupped the same flags in horizontal and in vertical?

File: t/cmd_pd [hunk 2]
```diff
@@ -662,7 +664,9 @@ pd 1 @ 0x004010fa
 e asm.cmtright=false
 pd 1 @ 0x004010fa
 '
-EXPECT='            0x004010fa      8b3500204000   mov esi, dword [0x402000]   ; ptr_GetLastError ; [0x402000:4]=0x29f4 reloc.KERNEL32.dll_GetLastError_244
+EXPECT='            0x004010fa      8b3500204000   mov esi, dword [0x402000]   ; section_end..text ; section..rdata ; ptr_GetLastError ; [0x402000:4]=0x29f4 reloc.KERNEL32.dll_GetLastError_244
```
Comment (radare) [inline t/cmd_pd]: same flags twice wat

File: t/cmd_pd [hunk 3]
```diff
@@ -646,7 +646,9 @@ pd 1 @ 0x004010fa
 e asm.cmtright=false
 pd 1 @ 0x004010fa
 '
-EXPECT='            0x004010fa      8b3500204000   mov esi, dword sym.imp.KERNEL32.dll_GetLastError ; [0x402000:4]=0x29f4 reloc.KERNEL32.dll_GetLastError_244
+EXPECT='            0x004010fa      8b3500204000   mov esi, dword sym.imp.KERNEL32.dll_GetLastError ; section_end..text ; section..rdata ; [0x402000:4]=0x29f4 reloc.KERNEL32.dll_GetLastError_244
```
Comment (radare) [inline t/cmd_pd]: more dupes

**Comment by kazarmy** [comment]
The tests look better now, right?

**Comment by radare** [comment]
Nope. I dont see why hiding section flags and showing the others twice is anything good here :?

**Comment by kazarmy** [comment]
So you want to:
1.  Show the section flags.
2.  Show either a string or its associated `str.` flag, but not both. The string is preferred, with the flag only shown if the string is not printable.
3.  Do something about the `reloc.` comments.

?

**Comment by kazarmy** [comment]
Okay, closing.

**Pull request closed without merging**